praw/models/reddit/emoji.py

```diff
@@ -128,28 +184,45 @@ def __iter__(self) -> List[Emoji]:
                print(emoji)
 
         """
-        response = self.subreddit._reddit.get(
+        response = self._reddit.get(
             API_PATH["emoji_list"].format(subreddit=self.subreddit)
         )
-        for emoji_name, emoji_data in response[
-            self.subreddit.fullname
-        ].items():
+        subreddit_keys = [
+            key
+            for key in response.keys()
+            if key.startswith(self._reddit.config.kinds["subreddit"])
+        ]
```

Comment:
What does this verification check do?

Reply:
From what I've observed `response` contains global Reddit emoji, and the emoji for the requested subreddit. However, it's possible that Reddit could modify it again in the future to add another key, or somehow return results from more than one subreddit. Previously we used the full name to get the exact key desired, however, that requires an HTTP request to get the subreddit fullname. This code grabs whatever keys begin with the subreddit prefix (`t5_`) and goes from there.

Reply:
Won't this conflict if there are multiple subreddits used?

Reply:
See https://github.com/praw-dev/praw/pull/1130/files#r376698310